pilot/pkg/model/validation_test.go

```diff
@@ -2831,6 +2831,90 @@ func TestValidateOutlierDetection(t *testing.T) {
 	}
 }
 
+func TestValidateEnvoyFilter(t *testing.T) {
+	tests := []struct {
+		name  string
+		in    proto.Message
+		error string
+	}{
+		{name: "empty filters", in: &networking.EnvoyFilter{}, error: "missing filters"},
+
+		{name: "missing relativeTo", in: &networking.EnvoyFilter{
+			Filters: []*networking.EnvoyFilter_Filter{
+				{
+					InsertPosition: &networking.EnvoyFilter_InsertPosition{
+						Index: networking.EnvoyFilter_InsertPosition_AFTER,
+					},
+					FilterType:   networking.EnvoyFilter_Filter_NETWORK,
+					FilterName:   "envoy.foo",
+					FilterConfig: &types.Struct{},
+				},
+			},
+		}, error: "missing relativeTo"},
+
+		{name: "missing filter type", in: &networking.EnvoyFilter{
+			Filters: []*networking.EnvoyFilter_Filter{
+				{
+					InsertPosition: &networking.EnvoyFilter_InsertPosition{
+						Index: networking.EnvoyFilter_InsertPosition_FIRST,
+					},
+					FilterName:   "envoy.foo",
+					FilterConfig: &types.Struct{},
+				},
+			},
+		}, error: "missing filter type"},
+
+		{name: "missing filter name", in: &networking.EnvoyFilter{
+			Filters: []*networking.EnvoyFilter_Filter{
+				{
+					InsertPosition: &networking.EnvoyFilter_InsertPosition{
+						Index: networking.EnvoyFilter_InsertPosition_FIRST,
+					},
+					FilterType:   networking.EnvoyFilter_Filter_NETWORK,
+					FilterConfig: &types.Struct{},
+				},
+			},
+		}, error: "missing filter name"},
+
+		{name: "missing filter config", in: &networking.EnvoyFilter{
+			Filters: []*networking.EnvoyFilter_Filter{
+				{
+					InsertPosition: &networking.EnvoyFilter_InsertPosition{
+						Index: networking.EnvoyFilter_InsertPosition_FIRST,
+					},
+					FilterType: networking.EnvoyFilter_Filter_NETWORK,
+					FilterName: "envoy.foo",
+				},
+			},
+		}, error: "missing filter config"},
```

Comment:
optional: add bad type for coverage completeness

Reply:
Type is enum.

Reply:
i meant pass in some different proto other than *networking.EnvoyFilter to ValidateEnvoyFilter.